src/main/org/apache/tools/ant/taskdefs/optional/ssh/Scp.java

```diff
@@ -435,7 +435,8 @@ private String parseUri(final String uri) {
             throw new BuildException("no username was given.  Can't authenticate.");
         }
 
-        final int indexOfPath = uri.indexOf(':', indexOfAt + 1);
+        // For IPv6 address having more than one ":", it should look for the last occurrence
+        final int indexOfPath = uri.lastIndexof(':');
         if (indexOfPath == -1) {
```

Comment:
we should still ensure `indexOfPath` is bigger than `indexOfAt`. So maybe we change the if to not only throw if it is -1 but rather if it is less than or equal `indexOfAt` (equal could be the case if `indexOfAt` is -1 itself.

Reply:
I agree.. have made the required changes